http-agent/src/test/java/org/motechproject/http/agent/listener/HttpClientEventListenerTest.java

```diff
@@ -197,4 +198,18 @@ public void shouldReturnResponse() throws Throwable {
         ResponseEntity<?> responseEntity = httpClientEventListener.handleWithUserPasswordAndReturnType(motechEvent);
         assertEquals(responseEntity, exceptedResponseEntity);
     }
+
+    @Test
+    public void shouldParseUsernameAsNull() {

```

Comment:
It should invoke our actual method - handleWithUserPasswordAndReturnType() and verify the parameters are properly parsed.